zbench.zig

```diff
@@ -286,13 +300,54 @@ pub const Benchmark = struct {
                 progress_node.setEstimatedTotalItems(0);
                 progress_node.setCompletedItems(0);
                 progress.refresh();
+
+                // Sum the durations from the timings_ns array
+                var total_duration_ns: u64 = 0;
+                for (x.readings.timings_ns) |time_ns| {
+                    total_duration_ns += time_ns;
+                }
+
+                // Append the result with the total duration
+                const newResult = try Result.init(x.name, x.readings, total_duration_ns);
+                try results.append(newResult); // Append the successfully created Result
+
+                if (total_duration_ns < fastest) {
+                    fastest = total_duration_ns;
+                    fastest_name = x.name;
+                }
+
                 try x.prettyPrint(arena.allocator(), writer, true);
                 _ = arena.reset(.retain_capacity);
             },
         };
+
+        // Print summary
+        if (self.common_config.display_summary) {
+            try printSummary(writer, results.items, fastest, fastest_name);
+        }
     }
 };
 
+fn printSummary(writer: anytype, results: []const Result, fastest_ns: u64, fastest_name: []const u8) !void {
+    try writer.print("\x1b[1mSummary:\x1b[0m\n", .{});
+    try writer.print("{s} ran\n", .{fastest_name});
+
+    var last_index = results.len - 1;
+    for (results, 0..) |result, index| {
+        // Skip the fastest since it is being used as the baseline
+        if (std.mem.eql(u8, result.name, fastest_name)) continue;
+
+        const total_duration_f64 = @as(f64, @floatFromInt(result.total_duration_ns));
+        const fastest_duration_f64 = @as(f64, @floatFromInt(fastest_ns));
+
+        const times_slower = total_duration_f64 / fastest_duration_f64;
+
+        const connector_char = if (index == last_index) " └─" else " ├─";
```

Comment:
There's an edge case here which I mentioned in the last PR, if the fastest case is the last one then you get an output like:
```
fast function ran
  ├─ 16.49x times faster than medium fast function
  ├─ 243.27x times faster than slow function
```
One way to fix that is to have another counter `j` say, which is only incremented for the non-fastest items, and it's compared to `results.len - 2`.

Reply:
True good point! Added that 👍 forgot to bring this one over from the last review.